crates/libs/core/src/client/package_change/config.rs

```diff
@@ -0,0 +1,146 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+//! Handle callbacks for configuration package changes
+//! TODO: We probably should also provide a helpful callback to use in conjunction with the config-rs support (so that it processes configuration changes)
+use mssf_com::FabricRuntime::{
+    IFabricConfigurationPackageChangeHandler, IFabricConfigurationPackageChangeHandler_Impl,
+};
+
+use crate::runtime::config::ConfigurationPackage;
+
+use super::PackageChangeType;
+
+#[derive(Debug, Clone)]
+pub struct ConfigurationPackageChangeEvent {
+    pub change_type: PackageChangeType,
```

Comment:
Maybe change_type could be an enum.

Reply:
It is, in crates/libs/core/src/client/package_change/mod.rs
```
pub enum PackageChangeType {
    Addition,
    Removal,
    Modification,
}
```
Do you mean I could make ConfigurationPackageChangeEvent an enum with actual fields?? Yeah, I could
```
pub enum PackageChangeEvent<T> {
    Addition { next_package: T },
    Removal { next_package: T }
    Modification { previous_package: T, next_package: T},
}
```
That's what I do for diffs inside my library

Reply:
Yes. Enum with fields.